Use system-openssl in Makefile.
Use /usr/bin/wish in gitk.
Fix perl install locations.

diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -184,7 +184,7 @@
 bindir = $(prefix)/bin
 mandir = $(prefix)/share/man
 infodir = $(prefix)/share/info
-gitexecdir = $(prefix)/libexec/git-core
+gitexecdir = $(prefix)/lib/git
 sharedir = $(prefix)/share
 template_dir = $(sharedir)/git-core/templates
 htmldir=$(sharedir)/doc/git-doc
@@ -199,7 +199,7 @@
 
 # default configuration for gitweb
 GITWEB_CONFIG = gitweb_config.perl
-GITWEB_CONFIG_SYSTEM = /etc/gitweb.conf
+GITWEB_CONFIG_SYSTEM = $(sysconfdir)/gitweb.conf
 GITWEB_HOME_LINK_STR = projects
 GITWEB_SITENAME =
 GITWEB_PROJECTROOT = /pub/git
@@ -788,6 +788,9 @@
 	endif
 endif
 
+BASIC_CFLAGS += -I%p/lib/system-openssl/include -I%p/include
+BASIC_LDFLAGS += -L%p/lib/system-openssl/lib -L%p/lib
+
 ifdef NO_R_TO_GCC_LINKER
 	# Some gcc does not accept and pass -R to the linker to specify
 	# the runtime dynamic library path.
diff --git a/perl/Makefile b/perl/Makefile
--- a/perl/Makefile
+++ b/perl/Makefile
@@ -38,7 +38,7 @@
 	echo '	echo $(instdir_SQ)' >> $@
 else
 $(makfile): Makefile.PL ../GIT-CFLAGS
-	$(PERL_PATH) $< PREFIX='$(prefix_SQ)'
+	$(PERL_PATH) $< PREFIX='$(prefix_SQ)' LIB='%p/lib/perl5' INSTALLSITEMAN3DIR='%p/share/man/man3'
 endif
 
 # this is just added comfort for calling make directly in perl dir
